ImageGallery/src/org/sleuthkit/autopsy/imagegallery/datamodel/DrawableDB.java

```diff
@@ -763,20 +773,20 @@ private void insertOrUpdateFile(DrawableFile f, @Nonnull DrawableTransaction tr,
 
 
    /**
-     * Gets all data source object ids from datasources table
+     * Gets all data source object ids from datasources table, and their DrawableDbBuildStatusEnum
      *
-     * @return list of known data source object ids
+     * @return map of known data source object ids, and their db status
      */
-    public Set<Long> getDataSourceIds() throws TskCoreException {
+    public Map<Long, DrawableDbBuildStatusEnum> getDataSourceDbBuildStatus() throws TskCoreException {
         Statement statement = null;
         ResultSet rs = null;
-        Set<Long> ret = new HashSet<>();
+        Map<Long, DrawableDbBuildStatusEnum> map = new HashMap<>();
         dbReadLock();
         try {
             statement = con.createStatement();
-            rs = statement.executeQuery("SELECT ds_obj_id FROM datasources "); //NON-NLS
+            rs = statement.executeQuery("SELECT ds_obj_id, drawable_db_build_status FROM datasources "); //NON-NLS
             while (rs.next()) {
-                ret.add(rs.getLong(1));
+                map.put(rs.getLong(1), DrawableDbBuildStatusEnum.valueOf(rs.getString(2)));
```

Comment:
I strongly advise against using indices to get the values out of a result set.  It is easy to change the query and forget to change the indices leading to bugs.  In addition I am not sure if the column order is guaranteed to be consistent across JDBC drivers.